File: backend/src/main/java/br/ufpr/sept/so2/modules/academico/api/dto/PeriodoLetivoResponse.java
package br.ufpr.sept.so2.modules.academico.api.dto;

import br.ufpr.sept.so2.modules.academico.domain.PeriodoLetivo;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;

public record PeriodoLetivoResponse(
        UUID id,
        int ano,
        int semestre,
        LocalDate inicio,
        LocalDate fim,
        boolean ativo,
        OffsetDateTime createdAt,
        OffsetDateTime updatedAt,
        @JsonProperty("_links") Map<String, String> links
) {

    public static PeriodoLetivoResponse from(PeriodoLetivo periodo) {
        String base = "/academico/periodos/" + periodo.getId();
        return new PeriodoLetivoResponse(
                periodo.getId(),
                periodo.getAno(),
                periodo.getSemestre(),
                periodo.getInicio(),
                periodo.getFim(),
                periodo.isAtivo(),
                periodo.getCreatedAt(),
                periodo.getUpdatedAt(),
                Map.of(
                        "self", base,
                        "atualizar", base,
                        "excluir", base
                )
        );
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/academico/api/dto/PeriodoLetivoResponse.kt
package br.ufpr.sept.so2.modules.academico.api.dto

import br.ufpr.sept.so2.modules.academico.domain.PeriodoLetivo
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDate
import java.time.OffsetDateTime
import java.util.UUID

data class PeriodoLetivoResponse(
    val id: UUID,
    val ano: Int,
    val semestre: Int,
    val inicio: LocalDate,
    val fim: LocalDate,
    val ativo: Boolean,
    val createdAt: OffsetDateTime,
    val updatedAt: OffsetDateTime,
    @get:JsonProperty("_links")
    val links: Map<String, String>,
) {
    companion object {
        fun from(periodo: PeriodoLetivo): PeriodoLetivoResponse {
            val base = "/academico/periodos/${periodo.id}"
            return PeriodoLetivoResponse(
                id = periodo.id,
                ano = periodo.ano,
                semestre = periodo.semestre,
                inicio = periodo.inicio,
                fim = periodo.fim,
                ativo = periodo.ativo,
                createdAt = periodo.createdAt,
                updatedAt = periodo.updatedAt,
                links = mapOf(
                    "self" to base,
                    "atualizar" to base,
                    "excluir" to base,
                ),
            )
        }
    }
}
